azurerm/internal/services/azurestackhci/hci_cluster_resource_test.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/azurestackhci/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type HCIClusterResource struct{}
+
+func TestAccHCICluster_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_hci_cluster", "test")
+	r := HCIClusterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccHCICluster_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_hci_cluster", "test")
+	r := HCIClusterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccHCICluster_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_hci_cluster", "test")
+	r := HCIClusterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccHCICluster_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_hci_cluster", "test")
+	r := HCIClusterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r HCIClusterResource) Exists(ctx context.Context, client *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	clusterClient := client.AzureStackHCI.ClusterClient
+	id, err := parse.ClusterID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clusterClient.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+
+		return nil, fmt.Errorf("retrieving HCI Cluster %q: %+v", state.ID, err)
+	}
+
+	return utils.Bool(resp.ClusterProperties != nil), nil
+}
+
+func (r HCIClusterResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_hci_cluster" "test" {
+  name                = "acctest-HCICluster-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  client_id           = data.azurerm_client_config.current.client_id
+  tenant_id           = data.azurerm_client_config.current.tenant_id
```

Comment:
rather than using the current service principal, since this is expecting an AAD application can we spin one up and use that instead?

Reply:
Sorry.  For the other comments you mentioned, I've submitted PR. But for this comment, seems I have no permission to create an AAD application via "azuread_application". So I haven't updated test code against this case.

Reply:
you'll need to grant permissions to use [the Azure AD API](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/guides/service_principal_configuration) - unfortunately this has to provision an AD Application as a part of the test, reusing the SP would cause issues

Reply:
OK. Updated test code.